### PR TITLE
test: add check about OS auto detecting

### PR DESCRIPTION
**Pull request by yunmingyang**
When there is an OS already input, new "Installation source" could change the OS by detecting automatically

### CODE REVIEW

**Review by KKoukiou — CHANGES_REQUESTED** [review]
There is already a test - which at some point exactly what you 're trying to do here, but we broke it. 

Search for: # Test OS autodetection from URL tree media
https://github.com/cockpit-project/cockpit-machines/blob/main/test/check-machines-create#L75

So instead of introducing this new one - let's fix that one.

You can fix that we removing os_name=config.FEDORA_28, and adding a new expected_os_name which will just wait for the field to get filled.

Then you don't need that checkOsDetect function.

**Comment by yunmingyang** [comment]
OK, let me check and modify it later.

File: test/check-machines-create [hunk 1]
```diff
@@ -911,6 +928,17 @@ class TestMachinesCreate(VirtualMachinesCase):
             elif self.os_name:
                 b.click("#os-select-group > div button")
                 b.click("#os-select li:contains('{0}') button".format(self.os_name))
+                b.wait_attr("#os-select-group input", "value", self.os_name)
+                if (self.expected_os_name and
+                    self.expected_location and
+                    self.sourceType == 'url'):
+                    b.set_input_text("#source-url", self.expected_location)
```
Comment (KKoukiou) [inline test/check-machines-create:935]: Why don't you use self.location. expected_ prefix is used for when the dialog autofills that input field, in this case self.expected_os_name. The fields that we set in the test manually should not use expected.

File: test/check-machines-create [hunk 2]
```diff
@@ -911,6 +928,17 @@ class TestMachinesCreate(VirtualMachinesCase):
...
+                    b.wait_attr("#os-select-group", "data-loading", "false")
+                    b.wait_attr("#os-select-group input", "value", self.expected_os_name)
+
+                    b.switch_to_top()
```
Comment (KKoukiou) [inline test/check-machines-create:939]: This does not belong here. If you want to check that the page did not crash please introduce a function and run it after the fill method.

**Review by KKoukiou — CHANGES_REQUESTED** [review]

File: test/check-machines-create (hunk 1 above)
Comment (yunmingyang) [reply test/check-machines-create]: OK, then change it from expected_location to locationSecondInput

File: test/check-machines-create [hunk 3]
```diff
@@ -81,6 +81,17 @@ class TestMachinesCreate(VirtualMachinesCase):
                                                             start_vm=False,
                                                             pixel_test_tag="auto"))
 
+        # Test OS auto detection can change "Operatin system" field without Ooops
+        runner.cancelDialogTest(TestMachinesCreate.VmDialog(self, sourceType='url',
+                                                            location=config.VALID_URL,
```
Comment (KKoukiou) [inline test/check-machines-create:86]: I don't get why you want to use locationSecondInput new variable. Can't you just set directly the location to TREE_URL and let the dialog autodetect the OS?

File: test/check-machines-create [hunk 4]
```diff
@@ -81,6 +81,17 @@ class TestMachinesCreate(VirtualMachinesCase):
                                                             start_vm=False,
                                                             pixel_test_tag="auto"))
 
+        # Test OS auto detection can change "Operatin system" field without Ooops
```
Comment (KKoukiou) [inline test/check-machines-create:84]: s/Operatin/Operating

Comment (KKoukiou) [reply test/check-machines-create:84]: Do you have some bugzilla to attach in the test as comment? No needed just wondering.

File: test/check-machines-create (hunk 3 above)
Comment (yunmingyang) [reply test/check-machines-create:86]: Yeah, make sense, I just follow the step of BZ and forget that

File: test/check-machines-create (hunk 4 above)
Comment (yunmingyang) [reply test/check-machines-create:84]: Yeah, I missed it, this case is related with https://bugzilla.redhat.com/show_bug.cgi?id=1715409

File: test/check-machines-create [hunk 5]
```diff
@@ -908,9 +923,16 @@ class TestMachinesCreate(VirtualMachinesCase):
...
-            elif self.os_name:
-                b.click("#os-select-group > div button")
+
+            if self.os_name:
+                # After OS input already has a value, the expand button need selector with more details
+                b.click("#os-select-group > div button.pf-c-select__toggle-button")
                 b.click("#os-select li:contains('{0}') button".format(self.os_name))
+                b.wait_attr("#os-select-group input", "value", self.os_name)
+                if self.expected_os_name and self.sourceType == 'url':
+                    b.set_input_text("#source-url", self.location)
```
Comment (KKoukiou) [inline test/check-machines-create:933]: Sorry for the iteration again.
This set_input_text("#source-url", self.location) is not needed.

It's already done in https://github.com/cockpit-project/cockpit-machines/pull/398/commits/d608cceef38a94702ac55fec87f4bec1fc81ec3d#diff-b284ac42752f40597c4dd12d74e16f6ed82ad5e12d68f0437f242f5ae20bbaa5R916 a few lines above.

Same for the condition, you can drop the self.sourceType == 'url'.

The checks for expected_os_name is enough.

File: test/check-machines-create [hunk 6]
```diff
@@ -655,14 +667,16 @@ class TestMachinesCreate(VirtualMachinesCase):
         vmId = 0
 
         def __init__(self, test_obj, name=None,
-                     sourceType='file', sourceTypeSecondChoice=None, location='',
+                     sourceType='file', sourceTypeSecondChoice=None,
+                     location='',
```
Comment (KKoukiou) [inline test/check-machines-create:671]: Unrelated newline change

**Review by KKoukiou — CHANGES_REQUESTED** [review]

**Review by KKoukiou — APPROVED** [review]
